apps/key-manager/src/app/api/key/route.ts

```diff
@@ -0,0 +1,97 @@
+"use server";
+
+import { createHash } from "node:crypto";
+import { auth } from "@/auth";
+import { getCloudflareContext } from "@opennextjs/cloudflare";
+import { createId } from "@paralleldrive/cuid2";
+
+const getUserPrefix = (userId: string) => {
+  const hash = createHash("sha256");
+  const prefix = hash.update(userId).digest("base64url");
+
+  return prefix;
+};
+
+const createUserKeyHelper = async (userId: string) => {
+  const prefix = getUserPrefix(userId);
+  const type = "sk";
+  const uniqueId = createId();
+  const completeKey = `${prefix}:${type}:${uniqueId}`;
+
+  const ctx = await getCloudflareContext();
+  await ctx.env.API_KEYS.put(completeKey, '{"_type":"secret","resources":{"FUZZY_SEARCH":true}}');
+
+  return completeKey;
+};
+
+/**
+ * Returns the user's API key
+ *
+ * @param id user's id
+ * @return the user's api key if it exists, otherwise null
+ */
+const getUserKeyHelper = async (id: string) => {
+  const ctx = await getCloudflareContext();
+
+  const prefix = getUserPrefix(id);
+  const listResult = await ctx.env.API_KEYS.list({ prefix, limit: 1 });
```

Comment:
One user can have many keys. It's still good to limit the max # of keys returned because otherwise it can be unbearably slow.

ref: https://community.cloudflare.com/t/cloudflare-kv-100x-faster-with-limit/212434